src/core_atmosphere/dynamics/mpas_atm_time_integration.F

```diff
@@ -720,11 +723,24 @@ subroutine atm_srk3(domain, dt, itimestep, exchange_halo_group)
       call mpas_timer_start('physics_get_tend')
       rk_step = 1
       dynamics_substep = 1
+
+! apply random perturbation pattern to the tendency
+      tend_names(1) = "rucuten"
+      tend_names(2) = "rvcuten"
+      tend_names(3) = "rublten"
+      tend_names(4) = "rvblten"
+      call stochastic_physics_pattern_apply(domain, 4, tend_names, ierr)
```

Comment:
Are these the total U and V tendencies after all physics calls have been completed?  Why are there two sets of them?

Reply:
From Ning: Physics are combined in a "convective" and "BL" tendencies within MPAS.  One set of U and V is convective, the others is for the BL.